akka-actor-typed-tests/src/test/scala/akka/actor/typed/InterceptSpec.scala

```diff
@@ -35,9 +35,53 @@ object InterceptSpec {
       target(context, signal)
     }
 
-    override def isSame(other: BehaviorInterceptor[Any, Any]): Boolean =
+    override def isSame(other: BehaviorInterceptor[Any, Any, Any]): Boolean =
       other.isInstanceOf[SameTypeInterceptor]
   }
+
+  // This is similar to how EventSourcedBehavior is implemented
```

Comment:
This a good starting point that illustrates the problems. See new tests "Protocol transformer interceptor" below.